### ISSUE TITLE
How to use from typescript

### ISSUE THREAD

**Issue by RudolfVonKrugstein** [engineforce/ImmutableAssign#2]
I am trying to use immutable-assign like this from typescript:

```
import iassign = require('immutable-assign');
```

But I get this errors:

```
ERROR in [at-loader] node_modules\immutable-assign\src\iassign.ts:50:16 
    Cannot find name 'module'.

ERROR in [at-loader] node_modules\immutable-assign\src\iassign.ts:50:46 
    Cannot find name 'module'.

ERROR in [at-loader] node_modules\immutable-assign\src\iassign.ts:51:25 
    Cannot find name 'require'.

ERROR in [at-loader] node_modules\immutable-assign\src\iassign.ts:51:34 
    Cannot find name 'exports'.

ERROR in [at-loader] node_modules\immutable-assign\src\iassign.ts:51:65 
    Cannot find name 'module'.

ERROR in [at-loader] node_modules\immutable-assign\src\iassign.ts:71:25 
    Cannot find namespace 'DeepFreeze'.
```

**Comment by engineforce** [comment]
Looks like you are using awesome-typescript-loader and webpack that does not work well with my module; can you please give me some context, such as the minimal files in your project that can reproduce the problem. Thanks.

**Comment by RudolfVonKrugstein** [comment]
You are correct. I am using webpack with awesome-typescript-loader.

Here is the minimal file:

```
import iassign = require('immutable-assign');

let x = iassign([], (l) => {l.push(""); return l;})
```

And this is my webpack.config.js

```
module.exports = {
  entry: "./src/index.tsx",
  output: {
    filename: "bundle.js",
    path: __dirname + "/dist"
  },

  // Enable sourcemaps for debugging webpack's output.
  devtool: "source-map",

  resolve: {
    // Add '.ts' and '.tsx' as resolvable extensions.
    extensions: ["", ".webpack.js", ".web.js", ".ts", ".tsx", ".js"]
  },

  module: {
    loaders: [
      // All files with a '.ts' or '.tsx' extension will be handled by 'awesome-typescript-loader'.
        { test: /\.tsx?$/, loader: "awesome-typescript-loader" },
        { test: /\.css$/, loader: "style-loader!css-loader"}
    ],

    preLoaders: [
      // All output '.js' files will have any sourcemaps re-processed by 'source-map-loader'.
      { test: /\.js$/, loader: "source-map-loader" }
    ]
  },

  // When importing a module whose path matches one of the following, just
  // assume a corresponding global variable exists and use that instead.
  // This is important because it allows us to avoid bundling all of our
  // dependencies, which allows browsers to cache those libraries between builds.
  externals: {
    "react": "React",
    "react-dom": "ReactDOM"
  },
};
```

I use awesome-typescript-loader just because the guide for setting up a webpack/react project I read used it. Would you recommend something else (that works with immutable-assign)?

**Comment by engineforce** [comment]
The problem has been fixed in 1.0.26, please try.

**Comment by RudolfVonKrugstein** [comment]
It works. Thank you!
